elastic/beats
# libbeat/idxmgmt/ilm: fix alias creation

## What does this PR do?

When creating the initial index/write alias fails, don't check the status code, just check if the alias exists regardless of the error. The status code will be different (at least 400 and 500) depending on failure scenario.

## Why is it important?

This fixes a bug where alias creation fails when the alias exists but points to another index, and the initial index does not exist, e.g. due to ILM deletion.

## Checklist

~- [ ] I have made corresponding changes to the documentation~
~- [ ] I have made corresponding change to the default configuration files~

## How to test this PR locally

1. Run a beat with `setup.ilm.overwrite: true`
2. In Elasticsearch, force a rollover (`POST /<alias>/_rollover`) and delete the initial index
3. Restart the beat; there should be no errors due to creating the write alias

## Related issues

Closes #26142
